agent/util-scripts/test-bin/test-require-rpm

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# Test require-rpm via a series of simple invocations
+
+# The way we verify the behavior of require-rpm itself is by mocking out the
+# behavior of `rpm` and `rpmdev-vercmp` to behave the way we want.  The name
+# of the RPM encodes the behavior.
+
+_tmp=$(dirname "${_testout}")/tmp
+function _cleanup {
+    rm -rf ${_tmp}
+}
+trap _cleanup INT QUIT TERM EXIT
+mkdir ${_tmp}
+if [[ ${?} -ne 0 ]]; then
+    printf -- "test-require-rpm: failed to create temporary directory, %s\n" "${_tmp}" >&2
+    exit 1
+fi
+
+_prefix="Verify require-rpm with"
+function run_req_rpm {
+    _desc="${1}"
+    shift
+    printf -- "\n%s %s\n" "${_prefix}" "${_desc}"
```

Comment:
Just out of curiosity, why do you use `printf` here, instead of, say, `echo` like line 25, et al.?
```
    echo -e "\n${_prefix} ${_desc}\n"
```

Reply:
When using `bash` I try to maintain a habit of not using `echo` and use `printf`.  In many cases it is no different, but it is easier to enhance the format string than to get the variable expansion correct as the complexity grows.

Or maybe because I really miss working with C (not joking, I always have a hard time with not knowing exactly what side effects statements have).